src/Microsoft.DotNet.Darc/DarcLib/Actions/Remote.cs

```diff
@@ -1486,9 +1486,10 @@ public async Task<SemanticVersion> GetToolsDotnetVersionAsync(string repoUri, st
     /// <returns>Async task</returns>
     public async Task AddAssetLocationToDependenciesAsync(IEnumerable<DependencyDetail> dependencies)
     {
+        var buildCache = new Dictionary<int, Build>();
```

Comment:
I am only wondering if asset.BuildId in `if (!buildCache.TryGetValue(asset.BuildId, out Build producingBuild))` is unique.. Besides that, seems you found the spot.